harness/tests/experiment/fixtures/pytorch_onevar_model.py

```diff
@@ -221,13 +235,92 @@ def build_validation_data_loader(self) -> torch.utils.data.DataLoader:
             raise ValueError(f"unknown dataloader_type: {self.hparams['dataloader_type']}")
 
 
-class OneVarTrialWithApexAmp(OneVarTrial):
+class OneVarApexAMPTrial(OneVarTrial):
     def __init__(self, context: pytorch.PyTorchTrialContext) -> None:
         super().__init__(context)
         self.model, self.optimizer = self.context.configure_apex_amp(
-            models=self.model, optimizers=self.opt
+            models=self.model,
+            optimizers=self.opt,
+            opt_level="O2",
         )
 
+    def train_batch(
+        self, batch: pytorch.TorchData, epoch_idx: int, batch_idx: int
+    ) -> Dict[str, torch.Tensor]:
+        metrics = super().train_batch(batch, epoch_idx, batch_idx)
+        metrics["scale"] = apex.amp.state_dict()["loss_scaler0"]["loss_scale"]
+        return metrics
+
+
+class OneVarAutoAMPTrial(OneVarTrial):
+    def __init__(self, context: pytorch.PyTorchTrialContext) -> None:
+        context.experimental.use_amp()
+        super().__init__(context)
+        self.scaler = context._scaler
+
+    def train_batch(
+        self, batch: pytorch.TorchData, epoch_idx: int, batch_idx: int
+    ) -> Dict[str, torch.Tensor]:
+        metrics = super().train_batch(batch, epoch_idx, batch_idx)
+        metrics["scale"] = self.scaler.get_scale()
+        return metrics
+
+
+class OneVarManualAMPTrial(OneVarTrial):
+    def __init__(self, context: pytorch.PyTorchTrialContext) -> None:
+        self.scaler = context.wrap_scaler(GradScaler())
+        super().__init__(context)
+
+    def train_batch(
+        self, batch: pytorch.TorchData, epoch_idx: int, batch_idx: int
+    ) -> Dict[str, torch.Tensor]:
+        data, label = batch
+
+        self.cls_reducer.update(sum(label), batch_idx)
+        self.fn_reducer.update((sum(label), batch_idx))
```

Comment:
the custom reducers here are from an orthogonal feature.  Does it work if you just don't update them, leaving them totally out of the subclass definition?

Reply:
Cool. Gone!